src/main/java/com/aws/greengrass/util/ProxyUtils.java

```diff
@@ -169,6 +194,10 @@ public static HttpProxyOptions getHttpProxyOptions(DeviceConfiguration deviceCon
         httpProxyOptions.setHost(ProxyUtils.getHostFromProxyUrl(proxyUrl));
         httpProxyOptions.setPort(ProxyUtils.getPortFromProxyUrl(proxyUrl));
 
+        if ("https".equalsIgnoreCase(getSchemeFromProxyUrl(proxyUrl))) {
+            httpProxyOptions.setTlsContext(tlsContext);
```

Comment:
why not create the context in here since this is where you know the pertinent information?

Reply:
I didn't want any potential resource leak issues so I incorporated creating/closing into the mqtt client lifecycle.